ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/IndentationRule.kt

```diff
@@ -480,13 +483,39 @@ class IndentationRule : Rule(
     private class IndentContext {
         private val exitAdj = mutableMapOf<ASTNode, Int>()
         val ignored = mutableSetOf<ASTNode>()
-        val blockOpeningLineStack: Deque<Int> = LinkedList()
+        val blockStack: Deque<Block> = LinkedList()
         var localAdj: Int = 0
+
         fun exitAdjBy(node: ASTNode, change: Int) {
             exitAdj.compute(node) { _, v -> (v ?: 0) + change }
         }
+
         fun clearExitAdj(node: ASTNode): Int? =
             exitAdj.remove(node)
+
+        data class Block(
+            // Element type used for opening the block
+            val openingElementType: IElementType,
+            // Line at which the block is opened
+            val line: Int,
+            // Type of indentation to be used for the block
+            val blockIndentationType: BlockIndentationType
+        ) {
+            enum class BlockIndentationType {
```

Comment:
Seems like a good change. And easy to add more types for other edge cases.

Reply:
Tnx but I truly do hope that we do not need more edge cases though ;-)